src/core/src/type/bfloat16.cpp

```diff
@@ -57,8 +57,10 @@ size_t bfloat16::size() const {
 }
 
 bfloat16::operator float() const {
-    uint32_t tmp = (static_cast<uint32_t>(m_value) << 16);
-    const float* f = reinterpret_cast<const float*>(&tmp);
+    uint32_t tmp = 0;
+    uint32_t* ptmp = &tmp;
+    *ptmp = (static_cast<uint32_t>(m_value) << 16);
+    const float* f = reinterpret_cast<const float*>(ptmp);
     return *f;
 }
 
```

Comment:
Ubuntu 22.04 gcc will check tmp is used unintialized.
Tryed kinds of modification, found only change to pointer work.

Reply:
Is it constrain comes from new gcc version?